week-8/Homework/mandatory/3-dog-photo-gallery/dogApi.js

```diff
@@ -0,0 +1,20 @@
+function getData(apiURL) {
+  fetch(apiURL)
+    .then((respone) => {
+      if (respone.ok) {
+        return respone.json();
+      } else {
+        throw new Error(`404 Page not Found :(`);
```

Comment:
This will throw a `404` error no matter what actually happened in the code which could get confusing later on.

To do this better I'd recommend adding a `Catch` to the fetch and then logging a normal general error.

You can find out how here:
https://stackoverflow.com/questions/38235715/fetch-reject-promise-and-catch-the-error-if-status-is-not-ok